src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/PosTransactionsOrderRule.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Consensus.Rules;
+
+namespace Stratis.Bitcoin.Features.Consensus.Rules.CommonRules
+{
+    /// <summary>Checks that coinstake tx from the proven header is the 2nd transaction in a block and that the block has at least 2 transactions.</summary>
+    public class PosTransactionsOrderRule : IntegrityValidationConsensusRule
```

Comment:
Is this the right name?
Looks like the main check of this Rule is that the coinstake on the block and on the proven header match.
There re is no regard to the order of transactions